ibridges/utils/path.py

```diff
@@ -45,25 +76,27 @@ def remove(self):
                 coll = self.session.irods_session.collections.get(str(self))
                 coll.remove()
             elif self.dataobject_exists():
-                obj = self.session.irods_session.data_objects.get(irods_path)
+                obj = self.session.irods_session.data_objects.get(str(self))
                 obj.unlink()
-        except irods.exception.CUT_ACTION_PROCESSED_ERR:
-            raise(irods.exception.CUT_ACTION_PROCESSED_ERR('iRODS server forbids action.'))
+        except irods.exception.CUT_ACTION_PROCESSED_ERR as exc:
+            raise irods.exception.CUT_ACTION_PROCESSED_ERR(
+                f"While removing {self}: iRODS server forbids action.") from exc
 
     @staticmethod
     def create_collection(session,  coll_path: str) -> irods.collection.iRODSCollection:
         """Create a collection and all collections in its path. Return he collection. If the
         collection already exists, return ir.
-        
+
         Return
         ------
         irods.collection.iRODSCollection
-            The 
+            The
         """
         try:
             return session.irods_session.collections.create(str(coll_path))
-        except irods.exception.CUT_ACTION_PROCESSED_ERR:
-            raise(irods.exception.CUT_ACTION_PROCESSED_ERR('iRODS server forbids action.'))
+        except irods.exception.CUT_ACTION_PROCESSED_ERR as exc:
+            raise irods.exception.CUT_ACTION_PROCESSED_ERR(
+                "While creating collection '{coll_path}': iRODS server forbids action.") from exc
 
     def rename(self):
```

Comment:
The method is not implemented yet. Maybe we can print that to the prompt or throw a warning.

Reply:
Same as with the walk method.